cmd/kubeadm/app/cmd/phases/certs.go

```diff
@@ -66,6 +67,23 @@ var (
 		` + cmdutil.AlphaDisclaimer)
 )
 
+// certsData defines the behavior that a runtime data struct passed to the kubeconfig phase should
+// have. Please note that we are using an interface in order to make this phase reusable in different workflows
+// (and thus with different runtime data struct, all of them requested to be compliant to this interface)
+type certsData interface {
+	Cfg() *kubeadmapi.InitConfiguration
+}
+
+// NewCertsPhase returns the phase for the certs
+func NewCertsPhase() workflow.Phase {
+	return workflow.Phase{
+		Name:   "certs",
+		Short:  "Certificate generation",
+		Long:   "Certificate generation for the whole cluster",
+		Phases: getCertsSubPhases(),
```

Comment:
The UX generation for subcommands is kind of fragmented: `getCertsSubPhases`, `makePhase`, `makePhaseForCert`, `makePhasesForCA`
IMO, as soon as run functions will be moved into functions as suggested below, the above set of functions can be simplified...

Reply:
Updated